### PR TITLE
Add changelog for 1.5.0-RC1

### PR DESCRIPTION
**Pull request by techknowlogick**
As title

~Blocked by #4323~

### CODE REVIEW

**Review by sapk — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,267 @@ This changelog goes through all the changes that have been made in each release
...
+  * Testing using go 1.10 version. (#3899)
+  * Pass secrets to drone build step for docker (#3879)
+  * Add swagger check (#3811)
+  * Docker multi-stage (#2927)
+  * Simplify docker build process. (#2827)
+  * Snap-packaging for gitea (#2568)
+* DOCS
+  * Add details about USER_UID and USER_GID environment variables (#4301)
+  * Reorganized feature comparison (#4264)
+  * Fix #4255 - missing close tags in U2F (#4256)
```
Comment (daviian) [inline CHANGELOG.md]: regression from new feature

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,267 @@ This changelog goes through all the changes that have been made in each release
...
+  * Fix checklist padding (#3460)
+* ENHANCEMENT
+  * Add config path as an optionial flag when changing pass via CLI (#4184)
+  * Show only needed values in database configuration section depending on type (#4164)
+  * Mac OS X -> macOS (#4056)
+  * Make home.tmpl visible with Signin-View enabled (#4040)
+  * LFS: make HTTP auth period configurable (#4035)
+  * Add Environment Variables to Docker template (#4012)
+  * Fix typo in U2F description (#4007)
+  * Add missing token validation and missing alert in UI (#3976)
```
Comment (daviian) [inline CHANGELOG.md]: part of 1.4.2

File: CHANGELOG.md [hunk 3]
```diff
@@ -4,6 +4,267 @@ This changelog goes through all the changes that have been made in each release
...
+  * Update gitgraph.js to fix "Cannot read property color of undefined" (#4095)
+  * Fix #4090 by escaping filename page/img link (without path) for short link (#4091)
+  * Fix #4081 Check for leading / in base before removing it (#4082)
+  * Fix attachment ENABLED (#4064)
+  * Update git vendor (#4059)
+  * Fix relative markdown links with anchors (#4058)
+  * Added deletion of an empty line at the end of the file (#4050) (#4054)
+  * Hide issues from org private repos w/o team assignment (#4034)
+  * Fix writer cannot read bare repo guide (#4033)
+  * Fix Webhook and hook_task not cleaned up when delete repository (#4004)
```
Comment (daviian) [inline CHANGELOG.md]: part of 1.4.2

File: CHANGELOG.md [hunk 4]
```diff
@@ -4,6 +4,267 @@ This changelog goes through all the changes that have been made in each release
...
+  * Fix webhook bug of response info is not displayed in UI (#4000)
```
Comment (daviian) [inline CHANGELOG.md]: part of 1.4.2

File: CHANGELOG.md [hunk 5]
```diff
@@ -4,6 +4,267 @@ This changelog goes through all the changes that have been made in each release
...
+  * Remove dupelicate link (#4100)
+  * Update gitgraph.js to fix "Cannot read property color of undefined" (#4095)
+  * Fix #4090 by escaping filename page/img link (without path) for short link (#4091)
+  * Fix #4081 Check for leading / in base before removing it (#4082)
+  * Fix attachment ENABLED (#4064)
+  * Update git vendor (#4059)
+  * Fix relative markdown links with anchors (#4058)
+  * Added deletion of an empty line at the end of the file (#4050) (#4054)
+  * Hide issues from org private repos w/o team assignment (#4034)
+  * Fix writer cannot read bare repo guide (#4033)
```
Comment (daviian) [inline CHANGELOG.md]: part of 1.4.2

File: CHANGELOG.md [hunk 6]
```diff
@@ -4,6 +4,267 @@ This changelog goes through all the changes that have been made in each release
...
+  * Fix wiki redirects (#3919)
+  * consistent IsOverdue field name capitalization (#3909)
+  * Fix code tab link when viewing tags (#3908)
+  * Do not allow inactive users to access repositories using private keys (#3887)
+  * Add "error" as reserved username (#3882)
+  * Fix path cleanup in multiple places (#3871)
+  * Fix to use only needed columns from tables to get repository git paths (#3870)
+  * Fix only updated_unix when adding a comment (#3855)
+  * don't reset team/repo count when updating team via API (#3831)
+  * Don't force due date to current time (#3830)
```
Comment (daviian) [inline CHANGELOG.md]: part of 1.4.2

File: CHANGELOG.md [hunk 7]
```diff
@@ -4,6 +4,267 @@ This changelog goes through all the changes that have been made in each release
...
+  * fix links to prevent 404 after e.g. submitting a faulty form (#3982)
+  * Fix some webhooks bugs (#3981)
+  * Fix redirects on user settings introduced by menu refactoring (#3975)
+  * Fix error preventing profile settings update without selecting a language first (#3967)
+  * Fix blank topic on explore repo list (#3956)
+  * Fix blank topic (#3948)
+  * Adjust z-index for floating labels (#3939)
+  * Fix multiple asssingee table migration (#3931)
+  * Fixed issue deadline not showing on issue page (#3930)
+  * Fix wiki redirects (#3919)
```
Comment (daviian) [inline CHANGELOG.md]: part of 1.4.2

File: CHANGELOG.md [hunk 8]
```diff
@@ -4,6 +4,267 @@ This changelog goes through all the changes that have been made in each release
...
+  * Improve wiki content width. (#4195)
+  * HTML-escape plain-text READMEs in previews (#4192)
+  * Fix typo in OAUTH page (#4151)
+  * Only show "You cannot fork a repository you own" when needed. (#4130)
+  * Removed unnecessary line referencing LFS struct (#4113)
+  * Remove dupelicate link (#4100)
+  * Update gitgraph.js to fix "Cannot read property color of undefined" (#4095)
+  * Fix #4090 by escaping filename page/img link (without path) for short link (#4091)
+  * Fix #4081 Check for leading / in base before removing it (#4082)
+  * Fix attachment ENABLED (#4064)
```
Comment (daviian) [inline CHANGELOG.md]: part of 1.4.2

File: CHANGELOG.md [hunk 9]
```diff
@@ -4,6 +4,267 @@ This changelog goes through all the changes that have been made in each release
...
+  * Update git vendor (#4059)
+  * Fix relative markdown links with anchors (#4058)
+  * Added deletion of an empty line at the end of the file (#4050) (#4054)
```
Comment (daviian) [inline CHANGELOG.md]: part of 1.4.2

File: CHANGELOG.md [hunk 10]
```diff
@@ -4,6 +4,267 @@ This changelog goes through all the changes that have been made in each release
...
+  * Fix not removed watches on unallowed repositories (#4201)
+  * Improve wiki content width. (#4195)
+  * HTML-escape plain-text READMEs in previews (#4192)
+  * Fix typo in OAUTH page (#4151)
+  * Only show "You cannot fork a repository you own" when needed. (#4130)
+  * Removed unnecessary line referencing LFS struct (#4113)
+  * Remove dupelicate link (#4100)
+  * Update gitgraph.js to fix "Cannot read property color of undefined" (#4095)
+  * Fix #4090 by escaping filename page/img link (without path) for short link (#4091)
+  * Fix #4081 Check for leading / in base before removing it (#4082)
```
Comment (daviian) [inline CHANGELOG.md]: part of 1.4.2

File: CHANGELOG.md [hunk 11]
```diff
@@ -4,6 +4,267 @@ This changelog goes through all the changes that have been made in each release
...
+  * Fix blank topic on explore repo list (#3956)
+  * Fix blank topic (#3948)
+  * Adjust z-index for floating labels (#3939)
+  * Fix multiple asssingee table migration (#3931)
+  * Fixed issue deadline not showing on issue page (#3930)
+  * Fix wiki redirects (#3919)
+  * consistent IsOverdue field name capitalization (#3909)
+  * Fix code tab link when viewing tags (#3908)
+  * Do not allow inactive users to access repositories using private keys (#3887)
+  * Add "error" as reserved username (#3882)
```
Comment (daviian) [inline CHANGELOG.md]: part of 1.4.1

File: CHANGELOG.md [hunk 12]
```diff
@@ -4,6 +4,267 @@ This changelog goes through all the changes that have been made in each release
...
+  * Fix error preventing profile settings update without selecting a language first (#3967)
+  * Fix blank topic on explore repo list (#3956)
+  * Fix blank topic (#3948)
+  * Adjust z-index for floating labels (#3939)
+  * Fix multiple asssingee table migration (#3931)
+  * Fixed issue deadline not showing on issue page (#3930)
+  * Fix wiki redirects (#3919)
+  * consistent IsOverdue field name capitalization (#3909)
+  * Fix code tab link when viewing tags (#3908)
+  * Do not allow inactive users to access repositories using private keys (#3887)
```
Comment (daviian) [inline CHANGELOG.md]: part of 1.4.1 and is Security

File: CHANGELOG.md [hunk 13]
```diff
@@ -4,6 +4,267 @@ This changelog goes through all the changes that have been made in each release
...
+  * Add "error" as reserved username (#3882)
+  * Fix path cleanup in multiple places (#3871)
```
Comment (daviian) [inline CHANGELOG.md]: part of 1.4.1

File: CHANGELOG.md [hunk 14]
```diff
@@ -4,6 +4,267 @@ This changelog goes through all the changes that have been made in each release
...
+  * Fix to use only needed columns from tables to get repository git paths (#3870)
+  * Fix only updated_unix when adding a comment (#3855)
+  * don't reset team/repo count when updating team via API (#3831)
+  * Don't force due date to current time (#3830)
+  * Fixed 0 closed issues not show 50% closed on activty page (#3802)
+  * Fix typo in deleteUser (#3781)
+  * Remove unnecessary Safe tags (#3778)
```
Comment (daviian) [inline CHANGELOG.md]: part of 1.4.1

File: CHANGELOG.md [hunk 15]
```diff
@@ -4,6 +4,267 @@ This changelog goes through all the changes that have been made in each release
...
+  * Using maintained spdx license list (#3484)
+  * Reduce repo indexer disk usage (#3452)
+  * Allow square brackets in external issue patterns (#3408)
+  * Enable caching on assets and avatars (#3376)
+  * Rework special link parsing in the post-processing of markup (#3354)
+  * Shows total tracked time in issue and milestone list (#3341)
+* SECURITY
+  * Fix open redirect vulnerability on login screen (#4312)
+  * Support secure cookie for csrf-token (#3839)
+  * Correctly check http git access rights for reverse proxy authorized users (#3721)
```
Comment (daviian) [inline CHANGELOG.md]: part of 1.4.1

File: CHANGELOG.md [hunk 16]
```diff
@@ -4,6 +4,267 @@ This changelog goes through all the changes that have been made in each release
...
+  * Adjust z-index for floating labels (#3939)
+  * Fix multiple asssingee table migration (#3931)
+  * Fixed issue deadline not showing on issue page (#3930)
+  * Fix wiki redirects (#3919)
+  * consistent IsOverdue field name capitalization (#3909)
+  * Fix code tab link when viewing tags (#3908)
+  * Do not allow inactive users to access repositories using private keys (#3887)
+  * Add "error" as reserved username (#3882)
+  * Fix path cleanup in multiple places (#3871)
+  * Fix to use only needed columns from tables to get repository git paths (#3870)
```
Comment (daviian) [inline CHANGELOG.md]: part 1.4.1

File: CHANGELOG.md [hunk 17]
```diff
@@ -4,6 +4,267 @@ This changelog goes through all the changes that have been made in each release
...
+  * Use the active branch for the code tab (#3720)
+  * Set default branchname on first push (#3715)
+  * Add your_repositories to types (#3699)
+  * Escape branch name in dropdown menu (#3691)
+  * Update markbates/goth libary to fix OAuth2 support (#3661)
+  * Fix MySQL and PostgreSQL column drop SQL (#3649)
+  * Fix column removal in MSSQL (#3638)
+  * Fix incorrect 'Erase Wiki Data' string identifier. (#3613)
+  * Fix add feed for organization (#3594)
+  * Fix gpg expired bug when time is zero (#3584)
```
Comment (daviian) [inline CHANGELOG.md]: 1.4.1

File: CHANGELOG.md [hunk 18]
```diff
@@ -4,6 +4,267 @@ This changelog goes through all the changes that have been made in each release
...
+  * Fix multiple asssingee table migration (#3931)
+  * Fixed issue deadline not showing on issue page (#3930)
+  * Fix wiki redirects (#3919)
+  * consistent IsOverdue field name capitalization (#3909)
+  * Fix code tab link when viewing tags (#3908)
+  * Do not allow inactive users to access repositories using private keys (#3887)
+  * Add "error" as reserved username (#3882)
+  * Fix path cleanup in multiple places (#3871)
+  * Fix to use only needed columns from tables to get repository git paths (#3870)
+  * Fix only updated_unix when adding a comment (#3855)
```
Comment (daviian) [inline CHANGELOG.md]: 1.4.1

File: CHANGELOG.md [hunk 19]
```diff
@@ -4,6 +4,267 @@ This changelog goes through all the changes that have been made in each release
...
+  * don't reset team/repo count when updating team via API (#3831)
+  * Don't force due date to current time (#3830)
+  * Fixed 0 closed issues not show 50% closed on activty page (#3802)
+  * Fix typo in deleteUser (#3781)
```
Comment (daviian) [inline CHANGELOG.md]: 1.4.1

File: CHANGELOG.md [hunk 20]
```diff
@@ -4,6 +4,267 @@ This changelog goes through all the changes that have been made in each release
...
+  * Remove unnecessary Safe tags (#3778)
+  * Show Clipboard button if disable HTTP of git protocol (#3773)
+  * Enhance healthcheck for database connectivity (#3754)
+  * Add tag check to release draft creation (#3729)
+  * Use the active branch for the code tab (#3720)
```
Comment (daviian) [inline CHANGELOG.md]: 1.4.1

File: CHANGELOG.md [hunk 21]
```diff
@@ -4,6 +4,267 @@ This changelog goes through all the changes that have been made in each release
...
+  * Set default branchname on first push (#3715)
```
Comment (daviian) [inline CHANGELOG.md]: 1.4.1

File: CHANGELOG.md [hunk 22]
```diff
@@ -4,6 +4,267 @@ This changelog goes through all the changes that have been made in each release
...
+  * Add "error" as reserved username (#3882)
+  * Fix path cleanup in multiple places (#3871)
+  * Fix to use only needed columns from tables to get repository git paths (#3870)
+  * Fix only updated_unix when adding a comment (#3855)
+  * don't reset team/repo count when updating team via API (#3831)
+  * Don't force due date to current time (#3830)
+  * Fixed 0 closed issues not show 50% closed on activty page (#3802)
+  * Fix typo in deleteUser (#3781)
+  * Remove unnecessary Safe tags (#3778)
+  * Show Clipboard button if disable HTTP of git protocol (#3773)
```
Comment (daviian) [inline CHANGELOG.md]: 1.4.1

**Review by daviian — CHANGES_REQUESTED** [review]

**Comment by lafriks** [comment]
I think changelog should be only generated from label changloge and milstone 1.5.0 to keep it clean:
https://github.com/go-gitea/gitea/issues?q=label%3Achangelog+is%3Aclosed+milestone%3A1.5.0

If anything else from this needs to be included label shoud be added to that PR

**Review by lafriks — CHANGES_REQUESTED** [review]
See my comment to have changelog cleaner and more useful

**Comment by techknowlogick** [comment]
@lunny conflict resolved
@daviian / @lafriks thanks for feedback. I've changed the list to only include the items with changelog label

**Comment by lafriks** [comment]
It is probably worth to review if there is anything else that is worth to mention in there and than label should be added to that issue

**Comment by axifive** [comment]
@lafriks, may be sorting enhancement https://github.com/go-gitea/gitea/pull/3969

**Comment by techknowlogick** [comment]
@axifive added.

@lafriks / @daviian please review changes

**Review by lafriks — APPROVED** [review]

**Comment by lafriks** [comment]
@daviian need your approval

**Review by appleboy — APPROVED** [review]

**Review by daviian — APPROVED** [review]

**Comment by sapk** [comment]
We can also add #4188 to indicate that binary can be checked with the public key. We could also indicate the public key fingerprint.

**Comment by daviian** [comment]
There are many more security fixes that didn't find it's way to a minor version of 1.4
These should be included as well.

Also I think there are further noticable PR's.
See https://github.com/go-gitea/gitea/pulls?utf8=%E2%9C%93&q=is%3Aclosed+milestone%3A1.5.0+is%3Apr+-label%3Akind%2Fdocs++-label%3Abackport%2Fv1.4+-label%3Akind%2Fbug+-label%3Akind%2Ftranslation+-label%3Achangelog
Of course not all of them, but some.

**Comment by lafriks** [comment]
@daviian I added changelog label to 11 more PR's from that list, just need to add them also to this PR

**Comment by lafriks** [comment]
@daviian @sapk done

**Pull request merged**